File: AG/AG_2_review/PC/PC.py
n=int(input())
for x in range(n):
    bk=input()
    itemN=int(input())
    maxCt=int(input())
    grid=[]
    for i in range(itemN):
        grid.append(int(input()))
    grid.sort(reverse=1)
    # print(grid)
    ans=0
    ckl=[grid[0]]
    for i in range(1,len(grid)):
        ck=1
        for j in range(len(ckl)):
            if ckl[j]+grid[i]<=maxCt:
                ans+=1
                ckl.pop(j)
                ck=0
                break
        if ck:
            ckl.append(grid[i])
    print(ans+len(ckl))
    if x!=n-1:
        print()

File: AG/AG_2_review/PC/PC.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    for (int x = 0; x < n; x++) {
        int itemN, maxCt;
        cin.ignore();
        string bk;
        getline(cin, bk);
        cin >> itemN >> maxCt;
        vector<int> grid(itemN);
        for (int i = 0; i < itemN; i++) {
            cin >> grid[i];
        }
        sort(grid.begin(), grid.end(), greater<int>());

        int ans = 0;
        vector<int> ckl;
        ckl.push_back(grid[0]);
        for (int i = 1; i < grid.size(); i++) {
            int ck = 1;
            for (int j = 0; j < ckl.size(); j++) {
                if (ckl[j] + grid[i] <= maxCt) {
                    ans++;
                    ckl.erase(ckl.begin() + j);
                    ck = 0;
                    break;
                }
            }
            if (ck) {
                ckl.push_back(grid[i]);
            }
        }
        cout << ans + ckl.size() << endl;
        if (x != n - 1) {
            cout << endl;
        }
    }
    return 0;
}